tachyon/math/finite_fields/generator/generator_util.cc

```diff
@@ -18,14 +20,42 @@ namespace tachyon::math {
 std::string MpzClassToString(const mpz_class& m) {
   size_t limb_size = gmp::GetLimbSize(m);
   if (limb_size == 0) {
-    return "UINT64_C(0)";
+    return "0";
   }
 
-  std::vector<std::string> ret;
-  ret.reserve(limb_size);
+  enum class Type {
+    kMaxIsInt,
+    kMaxIsUint32,
+    kMaxIsUint64,
+  };
+
+  Type type = Type::kMaxIsInt;
   for (size_t i = 0; i < limb_size; ++i) {
-    ret.push_back(absl::Substitute("UINT64_C($0)", gmp::GetLimb(m, i)));
+    mp_limb_t limb = gmp::GetLimb(m, i);
```

Comment:
I'm not sure what you are trying to do here. Shouldn't you have a vector of Type?

Reply:
What this code does is figure out the maximum type in the [std::initializer_list<T>](https://en.cppreference.com/w/cpp/utility/initializer_list). I'll leave a comment!

Reply:
Where is the initializer_list coming from?

Reply:
Also in `bazel-bin/tachyon/math/finite_fields/test/gf7_config.h`:

```c++
  // Before, it was generated with UINT64_C()
  // constexpr static uint32_t kSubgroupGenerator = UINT64_C(5);
  // After, it is generated with a proper type.
  // constexpr static uint32_t kSubgroupGenerator = 5;
```